Microsoft.Recognizers.Text.DateTime/Extractors/BaseSetExtractor.cs

```diff
@@ -105,16 +105,56 @@ public virtual List<Token> TimeEveryday(string text)
         public List<Token> MatchEach(IExtractor extractor, string text)
         {
             var ret = new List<Token>();
-            var ers = extractor.Extract(text);
-            foreach (var er in ers)
+            var matches = this.config.EachPrefixRegex.Matches(text);
+            foreach (Match match in matches)
             {
-                var beforeStr = text.Substring(0, er.Start ?? 0);
-                var match = this.config.EachPrefixRegex.Match(beforeStr);
-                if (match.Success && string.IsNullOrWhiteSpace(beforeStr.Substring(match.Index + match.Length)))
+                if (match.Success)
+                {
+                    var trimedText = text.Remove(match.Index, match.Length);
+                    var ers = extractor.Extract(trimedText);
+                    foreach (var er in ers)
+                    {
+                        // add some code for fixing some regex bugs of Spanish
+                        // in order to pass the test cases
+                        while (er.Text.StartsWith(" "))
+                        {
+                            er.Text = er.Text.Substring(1);
+                            er.Start = er.Start + 1;
+                            er.Length = er.Length - 1;
+                        }

```

Comment:
is this the same as trim? if yes, we should solve it in the baseextractor instead of post processing